lib/src/client.dart

```diff
@@ -481,18 +481,36 @@ class SubiquityClient {
   }
 
   Future<AdConnectionInfo> getActiveDirectory() async {
-    final request = await _openUrl('GET', 'active_directory');
-    return _receive(
-      'getActiveDirectory()',
-      request,
-      AdConnectionInfo.fromJson,
-      (method, response) => _formatResponseLog(
-        method,
-        AdConnectionInfo.fromJson(jsonDecode(response))
-            .hidePassword()
-            .toString(),
-      ),
-    );
+    try {
+      final request = await _openUrl('GET', 'active_directory');
+      return await _receive(
+        'getActiveDirectory()',
+        request,
+        (json) {
+          if (json == null || json is! Map<String, dynamic>) {
+            json = <String, dynamic>{};
+          }
+          return AdConnectionInfo.fromJson(json);
+        },
+        (method, response) {
+          if (response.isEmpty) {
+            throw Exception('Received empty response');
+          }
+          var decodedJson = jsonDecode(response);
+          if (decodedJson == null || decodedJson is! Map<String, dynamic>) {
+            decodedJson = <String, dynamic>{};
+          }
+          return _formatResponseLog(
+            method,
+            AdConnectionInfo.fromJson(decodedJson).hidePassword().toString(),
+          );
+        },
+      );
+    } catch (e) {
+      // Handle or rethrow the exception
+      print('Exception caught: $e');
+      rethrow;
+    }
```

Comment:
I'm not quite sure on why to catch and rethrow this exception instead of just letting it propagate. Could you clarify?

Reply:
You're right, I was initially doing more stuff here, but now it's dead code.